catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/federation/impl/SortedQueryMonitor.java

```diff
@@ -97,6 +102,11 @@ public SortedQueryMonitor(
 
   @Override
   public void run() {
+    if (originalThreadResources != null) {
```

Comment:
If we find we need to do this in several other places, we should consider abstracting Shiro behind one of our security classes.